beacon_node/network/src/router.rs

```diff
@@ -81,55 +86,69 @@ pub enum RouterMessage<T: EthSpec> {
 
 impl<T: BeaconChainTypes> Router<T> {
     /// Initializes and runs the Router.
+    #[allow(clippy::too_many_arguments)]
     pub fn spawn(
         beacon_chain: Arc<BeaconChain<T>>,
         network_globals: Arc<NetworkGlobals<T::EthSpec>>,
         network_send: mpsc::UnboundedSender<NetworkMessage<T::EthSpec>>,
         executor: task_executor::TaskExecutor,
         invalid_block_storage: InvalidBlockStorage,
+        beacon_processor_send: BeaconProcessorSend<T::EthSpec>,
+        beacon_processor_reprocess_tx: mpsc::Sender<ReprocessQueueMessage>,
         log: slog::Logger,
     ) -> error::Result<mpsc::UnboundedSender<RouterMessage<T::EthSpec>>> {
         let message_handler_log = log.new(o!("service"=> "router"));
         trace!(message_handler_log, "Service starting");
 
         let (handler_send, handler_recv) = mpsc::unbounded_channel();
 
-        let (beacon_processor_send, beacon_processor_receive) =
-            mpsc::channel(MAX_WORK_EVENT_QUEUE_LEN);
-
         let sync_logger = log.new(o!("service"=> "sync"));
+        // generate the message channel
+        let (sync_send, sync_recv) = mpsc::unbounded_channel::<SyncMessage<T::EthSpec>>();
+
+        let network_beacon_processor = NetworkBeaconProcessor {
+            beacon_processor_send,
+            duplicate_cache: DuplicateCache::default(),
+            chain: beacon_chain.clone(),
+            network_tx: network_send.clone(),
+            sync_tx: sync_send.clone(),
+            reprocess_tx: beacon_processor_reprocess_tx,
+            network_globals: network_globals.clone(),
+            invalid_block_storage,
+            executor: executor.clone(),
+            log: log.clone(),
+        };
+        let network_beacon_processor = Arc::new(network_beacon_processor);
+
+        // spawn the delayed lookup service
+        let (delayed_lookups_send, delayed_lookups_recv) =
+            mpsc::channel::<DelayedLookupMessage>(DELAY_QUEUE_CHANNEL_SIZE);
+        delayed_lookup::spawn_delayed_lookup_service(
+            &executor,
+            beacon_chain.clone(),
+            delayed_lookups_recv,
+            sync_send.clone(),
+            log.clone(),
+        );
```

Comment:
Move `spawn_delayed_lookup_service` here as `sync_send` no longer needs to be passed to sync manager.